Tests/LoggingTests/LoggingTest.swift

```diff
@@ -152,8 +152,8 @@ class LoggingTest: XCTestCase {
                                                    "lazy": .stringConvertible(LazyMetadataBox { "rendered-at-first-use" })])
     }
 
-    private func dontEvaluateThisString(file: StaticString = (#file), line: UInt = #line) -> Logger.Message {
-        XCTFail("should not have been evaluated", file: file, line: line)
+    private func dontEvaluateThisString(file: StaticString = #file, line: UInt = #line) -> Logger.Message {
+        XCTFail("should not have been evaluated", file: (file), line: line)
```

Comment:
so the issue is that the previous impl would still cause warnings, and we have to silence the warning when using the functions, rather when defining them?

Reply:
@ktoso no, worse, they would default to `LoggingTest.swift` instead of the caller's `#file`.

If you do `file: StaticString = #file`, then `#file` gets evaluated in the _caller_'s file. If you do `file: StaticString = (#file)`, then `(#file)` gets evaluated to the file where you type `(#file)`. It's totally broken...

Reply:
omg 😱

Reply:
but there's hope: https://forums.swift.org/t/revisiting-the-source-compatibility-impact-of-se-0274-concise-magic-file-names/37720

Reply:
Yeah, keeping an eye on there... I didn't fully get that it's as broken as it is at first.